### PR TITLE
Add needles to bower.json

### PR DESCRIPTION
**Pull request by moifort**
* Needle to add dependency
* Needle to add override configuration

### CODE REVIEW

File: app/templates/_bower.json [hunk 1]
```diff
@@ -38,6 +38,7 @@
         "sockjs-client": "1.0.3",
         "stomp-websocket": "2.3.4",
         <%_ } _%>
+        "swagger-ui": "jhipster-needle-bower-add-dependency - JHipster will add additional bower dependency",
```
Comment (jdubois) [inline app/templates/_bower.json]: Using "swagger-ui" has key doesn't look good to me

File: script-base.js [hunk 2]
```diff
@@ -125,6 +125,59 @@ Generator.prototype.addElementTranslationKey = function(key, value, language) {
...
+    try {
+        jhipsterUtils.rewriteFile({
+            file: fullPath,
+            needle: 'jhipster-needle-bower-add-override',
+            splicable: [
+                override
+            ]
+        });
+    } catch (e) {
+        console.log('\nUnable to find '.yellow + fullPath + '. Reference to '.yellow + 'bower override (bowerPackageName: ' + name + ', main:' + JSON.stringify(main) + ', ignore:' + isIgnored + ')' +' not added.\n'.yellow);
```
Comment (jdubois) [inline script-base.js]: See #2474 for not using the .yellow

File: app/templates/_bower.json (hunk 1 above)
Comment (moifort) [inline app/templates/_bower.json]: I know, but it was the only way to add the needle without breaking the bower install.
- I try to add in ignore package, but still have an error
- I find an other solution, to add `"ignoredDependencies": ["jhipster-needle-bower-add-dependency"]` in `.bowerrc`, it's work when you execute the `bower install`, but t's not working with `grunt:wiredep` :(

The only solution it to use an existing dependency and put before.

Or create an specific package like `jhipster-needle` in bower?

**Comment by andidev** [comment]
Do we really need needles for bower.json? Can't we just read and parse the json and add it programmatically and then write the json object into file?

**Comment by jdubois** [comment]
yes I think @andidev you have a good point

**Comment by moifort** [comment]
@andidev :+1:

**Comment by moifort** [comment]
Thanks @andidev it was a very good idea!

**Comment by moifort** [comment]
@Julien I resolve the merge conflict

File: script-base.js [hunk 3]
```diff
@@ -268,7 +315,7 @@ Generator.prototype.addGradlePlugin = function (group, name, version) {
             file: fullPath,
             needle: 'jhipster-needle-gradle-buildscript-dependency',
             splicable: [
-                'classpath group: \'' + group + '\', name: \'' + name + '\', version: \'' + version + '\''
+                    'classpath group: \'' + group + ', name: \'' + name + ', version: \'' + version + '\''
```
Comment (pascalgrimaud) [inline script-base.js]: can you review this ? see https://github.com/jhipster/generator-jhipster/pull/2479

File: script-base.js [hunk 4]
```diff
@@ -291,7 +338,7 @@ Generator.prototype.addGradleDependency = function (scope, group, name, version)
             file: fullPath,
             needle: 'jhipster-needle-gradle-dependency',
             splicable: [
-                scope + ' group: \'' + group + '\', name: \'' + name + '\', version: \'' + version + '\''
+                    scope + ' group: \'' + group + ', name: \'' + name + ', version: \'' + version + '\''
```
Comment (pascalgrimaud) [inline script-base.js]: can you review this ? see https://github.com/jhipster/generator-jhipster/pull/2479

**Pull request merged**